lilypad/server/api/v0/auth/github_api.py

```diff
@@ -0,0 +1,115 @@
+"""The `/github` API router."""
+
+from typing import Annotated
+
+import httpx
+from fastapi import APIRouter, Depends, HTTPException, Query, status
+from sqlmodel import Session, select
+
+from ...._utils import create_jwt_token
+from ....db import get_session
+from ....models import (
+    OrganizationPublic,
+    OrganizationTable,
+    UserOrganizationTable,
+    UserPublic,
+    UserRole,
+    UserTable,
+)
+from ....services import DeviceCodeService
+from ....settings import Settings, get_settings
+
+github_router = APIRouter()
+
+
+@github_router.get("/github/callback", response_model=UserPublic)
```

Comment:
question: this is just for `lilypad auth` right? we don't use this on the client?

Reply:
its used for both. if user does lilypad auth for the first time without an account, it would hit the callback endpoint. the 2nd use case is if the user is already logged in, we redirect directly to the projects page, so lilypad auth needs to handle it there too